File: backend/app/schemas/collaboration_analytics.py
import uuid
import datetime
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any

class PeriodComparison(BaseModel):
    current: int
    previous: int
    delta: int
    growth_rate: float

class CollaborationOverviewResponse(BaseModel):
    workspace_id: uuid.UUID
    time_window: str
    active_users: int
    total_members: int
    active_teams: int
    active_projects: int
    total_comments: int
    root_comments: int
    total_replies: int
    total_mentions: int
    notifications_generated: int
    notifications_read: int
    total_activities: int
    engagement_rate: float
    health_status: str # HEALTHY, MODERATE, LOW
    activity_growth: PeriodComparison
    comment_growth: PeriodComparison

class TeamAnalyticsItem(BaseModel):
    team_id: uuid.UUID
    team_name: str
    member_count: int
    active_members: int
    comment_count: int
    activity_count: int
    engagement_rate: float
    health_status: str

class TeamAnalyticsListResponse(BaseModel):
    total: int
    page: int = 1
    page_size: int = 50
    teams: List[TeamAnalyticsItem] = Field(default_factory=list)

class ProjectAnalyticsItem(BaseModel):
    project_id: uuid.UUID
    project_name: str
    member_count: int
    active_members: int
    resource_count: int
    comment_count: int
    reply_count: int
    activity_count: int
    engagement_rate: float

class ProjectAnalyticsListResponse(BaseModel):
    total: int
    page: int = 1
    page_size: int = 50
    projects: List[ProjectAnalyticsItem] = Field(default_factory=list)

class ActivityTimeSeriesPoint(BaseModel):
    date: str
    count: int
    by_type: Dict[str, int] = Field(default_factory=dict)

class ActivityAnalyticsResponse(BaseModel):
    time_window: str
    total_activities: int
    series: List[ActivityTimeSeriesPoint] = Field(default_factory=list)

class CommentAnalyticsResponse(BaseModel):
    total_comments: int
    root_comments: int
    replies: int
    reply_to_root_ratio: float
    avg_comments_per_project: float

class TopMentionedUser(BaseModel):
    user_id: uuid.UUID
    username: str
    mention_count: int

class MentionAnalyticsResponse(BaseModel):
    total_mentions: int
    unique_mentioned_users: int
    top_mentioned: List[TopMentionedUser] = Field(default_factory=list)

class NotificationAnalyticsResponse(BaseModel):
    total_generated: int
    total_read: int
    total_unread: int
    read_rate: float
    by_type: Dict[str, int] = Field(default_factory=dict)

class ResourceCollaborationResponse(BaseModel):
    total_linked_resources: int
    by_type: Dict[str, int] = Field(default_factory=dict)
    commented_resources: int

class TopContributorItem(BaseModel):
    user_id: uuid.UUID
    username: str
    activity_count: int
    comment_count: int
    mention_count: int

class TopContributorsResponse(BaseModel):
    contributors: List[TopContributorItem] = Field(default_factory=list)

File: frontend/src/api/collaborationAnalytics.ts
import { request } from './client';

export interface PeriodComparison {
  current: number;
  previous: number;
  delta: number;
  growth_rate: number;
}

export interface CollaborationOverview {
  workspace_id: string;
  time_window: string;
  active_users: number;
  total_members: number;
  active_teams: number;
  active_projects: number;
  total_comments: number;
  root_comments: number;
  total_replies: number;
  total_mentions: number;
  notifications_generated: number;
  notifications_read: number;
  total_activities: number;
  engagement_rate: number;
  health_status: 'HEALTHY' | 'MODERATE' | 'LOW';
  activity_growth: PeriodComparison;
  comment_growth: PeriodComparison;
}

export interface TeamAnalyticsItem {
  team_id: string;
  team_name: string;
  member_count: number;
  active_members: number;
  comment_count: number;
  activity_count: number;
  engagement_rate: number;
  health_status: string;
}

export interface TeamAnalyticsListResponse {
  total: number;
  page: number;
  page_size: number;
  teams: TeamAnalyticsItem[];
}

export interface ProjectAnalyticsItem {
  project_id: string;
  project_name: string;
  member_count: number;
  active_members: number;
  resource_count: number;
  comment_count: number;
  reply_count: number;
  activity_count: number;
  engagement_rate: number;
}

export interface ProjectAnalyticsListResponse {
  total: number;
  page: number;
  page_size: number;
  projects: ProjectAnalyticsItem[];
}

export interface ActivityTimeSeriesPoint {
  date: string;
  count: number;
  by_type: Record<string, number>;
}

export interface ActivityAnalyticsResponse {
  time_window: string;
  total_activities: number;
  series: ActivityTimeSeriesPoint[];
}

export interface TopContributorItem {
  user_id: string;
  username: string;
  activity_count: number;
  comment_count: number;
  mention_count: number;
}

export interface TopContributorsResponse {
  contributors: TopContributorItem[];
}

export async function getCollaborationOverview(timeWindow = '7d'): Promise<CollaborationOverview> {
  return request<CollaborationOverview>(`/collaboration/analytics/overview?time_window=${encodeURIComponent(timeWindow)}`);
}

export async function getTeamAnalytics(params?: { page?: number; page_size?: number; search?: string }): Promise<TeamAnalyticsListResponse> {
  const q = new URLSearchParams();
  if (params?.page) q.append('page', params.page.toString());
  if (params?.page_size) q.append('page_size', params.page_size.toString());
  if (params?.search) q.append('search', params.search);
  const qStr = q.toString() ? `?${q.toString()}` : '';
  return request<TeamAnalyticsListResponse>(`/collaboration/analytics/teams${qStr}`);
}

export async function getProjectAnalytics(params?: { page?: number; page_size?: number; search?: string }): Promise<ProjectAnalyticsListResponse> {
  const q = new URLSearchParams();
  if (params?.page) q.append('page', params.page.toString());
  if (params?.page_size) q.append('page_size', params.page_size.toString());
  if (params?.search) q.append('search', params.search);
  const qStr = q.toString() ? `?${q.toString()}` : '';
  return request<ProjectAnalyticsListResponse>(`/collaboration/analytics/projects${qStr}`);
}

export async function getActivityAnalytics(timeWindow = '7d'): Promise<ActivityAnalyticsResponse> {
  return request<ActivityAnalyticsResponse>(`/collaboration/analytics/activity?time_window=${encodeURIComponent(timeWindow)}`);
}

export async function getTopContributors(limit = 10): Promise<TopContributorsResponse> {
  return request<TopContributorsResponse>(`/collaboration/analytics/top-contributors?limit=${limit}`);
}
